nmoscommon/mdns/mdnsEngine.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+import zeroconf
+from nmoscommon.logger import Logger
+from .dnsListener import DNSListener
+from .mdnsCallbackHandler import MDNSAdvertisementCallbackHandler
+from .mdnsRegistration import MDNSRegistration
+from .mdnsExceptions import ServiceAlreadyExistsException
+from .mdnsInterfaceController import MDNSInterfaceController
+from .mdnsRegistrationController import MDNSRegistrationController
+from .mdnsSubscriptionController import MDNSSubscriptionController
+
+
+class MDNSEngine(object):
+
+    def start(self):
+        self.subscriptionController = MDNSSubscriptionController()
+        self.registrationController = MDNSRegistrationController()
+        self.interfaceController = MDNSInterfaceController()
+        self.logger = Logger('mdns-engine')
+
+    def stop(self):
+        self.close()
+
+    def close(self):
+        self.subscriptionController.close()
+        self.registrationController.close()
+        self.interfaceController.close()
+
+    def run(self):
+        pass
+
+    def register(self, name, regtype, port, txtRecord="", callback=None, interfaceIps=None):
+        if not interfaceIps:
+            interfaceIps = []
+        interfaces = self.interfaceController.getInterfaces(interfaceIps)
+        registration = MDNSRegistration(interfaces, name, regtype, port, txtRecord)
+        callbackHandler = MDNSAdvertisementCallbackHandler(callback, registration)
+        try:
+            self.registrationController.addRegistration(registration)
+        except zeroconf.NonUniqueNameException:
+            callbackHandler.entryColission()
+        except ServiceAlreadyExistsException:
+            callbackHandler.entryColission()
+        except zeroconf.Error as e:
+            callbackHandler.entryFailed()
+        else:
+            callbackHandler.entryEstablisted()
+
+    def update(self, name, regtype, txtRecord=None):
+        self.unregister(name, regtype)
+        self.register(name, type, txtRecord=txtRecord)
+
+    def unregister(self, name, regType):
+        self.registrationController.removeRegistration(name, regType)
+
+    def callback_on_services(self, regtype, callback, registerOnly=False, domain=None):
```

Comment:
Worth noting that it looks like the default for registerOnly has changed here. To be honest I'm slightly surprised we ever had it as a variable, but perhaps something requires it.